lib/joined.rb

```diff
@@ -35,7 +35,7 @@ def joined(oxford: true, words_connector: ', ', last_word_connector: ', and ', c
     final_connector.sub!(/^,/, '') unless oxford && length > 2
 
     result = "#{self[0...-1].join(words_connector)}#{final_connector}#{self[-1]}"
-    return result.gsub(/"([^"]+)"\s*,/, '"\1,"') if comma_before
```

Comment:
Introducing a return here creates an unnecessary imperative that any additional logic is added before, not after. 

I generally avoid multiple return paths like this, unless they are definitively final (such as in the guard clauses above).